src/net/cruhland/axioms/CoreAlgebra/Monoid.agda

```diff
@@ -0,0 +1,20 @@
+module net.cruhland.axioms.CoreAlgebra.Monoid where
+
+import net.cruhland.axioms.AbstractAlgebra as AA
+open import net.cruhland.axioms.Eq as Eq using (_≃_; Eq)
+  
+record Monoid (G : Set) : Set₁ where
+  infix 7 _⊙_
+  field
+    {{eq}} : Eq G  -- includes reflexive, symmetric, and transitive properties
```

Comment:
I recommend moving this into the record parameters:

```agda
record Monoid (G : Set) {{_ : Eq G}} : Set₁ where
```

This is because `Eq G` is usually only a property of `G`, and doesn't change with different `Monoid G` implementations. And if you already have an instance of `Eq G` defined somewhere, and then try to use an instance of `Monoid G` which has an `{{eq}} : Eq G` field, the two `Eq` instances will conflict (even if they resolve to the same implementation). Putting it into the parameters will avoid introducing a conflicting instance.